client/gutenberg/extensions/mailchimp/edit.jsx

```diff
@@ -1,44 +1,228 @@
-/** @format */
-
 /**
  * External dependencies
  */
-import { ServerSideRender, TextControl } from '@wordpress/components';
-import { InspectorControls } from '@wordpress/editor';
+import apiFetch from '@wordpress/api-fetch';
+import { __ } from 'gutenberg/extensions/presets/jetpack/utils/i18n';
+import {
+	Button,
+	ExternalLink,
+	PanelBody,
+	Placeholder,
+	Spinner,
+	TextControl,
+	withNotices,
+} from '@wordpress/components';
+import { InspectorControls, RichText } from '@wordpress/editor';
 import { Fragment, Component } from '@wordpress/element';
+
 /**
  * Internal dependencies
  */
+import { icon } from '.';
+
+const API_STATE_LOADING = 0;
+const API_STATE_CONNECTED = 1;
+const API_STATE_NOTCONNECTED = 2;
 
-import { fields } from '.';
+const NOTIFICATION_PROCESSING = 'processing';
+const NOTIFICATION_SUCCESS = 'success';
+const NOTIFICATION_ERROR = 'error';
 
 class MailchimpSubscribeEdit extends Component {
-	constructor( ...args ) {
-		super( ...args );
-		this.fields = fields.map( field => {
-			field.set = value => this.props.setAttributes( { [ field.id ]: value } );
-			return field;
-		} );
+	constructor() {
+		super( ...arguments );
+		this.state = {
+			audition: null,
+			connected: API_STATE_LOADING,
+			connectURL: null,
+		};
+		this.timeout = null;
 	}
-
-	render() {
-		const { attributes } = this.props;
+	componentDidMount = () => {
+		this.apiCall();
+	};
+	onError = ( code, message ) => {
```

Comment:
This only seems to be used here once with a `null` code. Should we change the signature to just take a message?

Reply:
Sure, good call.

Reply:
017ba6976a1ffb91bfcf56823698cc396f901cc2